lib/templates/lab2/lab2.dart

```diff
@@ -1,19 +1,62 @@
-class Book {
-  // your code here
+class Book with JsonSerializable {
+  final String title;
+  final String author;
+  final int numberOfPages;
+  
+  Book(this.title, this.author, this.numberOfPages);
+
+  void bookDetails(){
+    print('$title, $author, $numberOfPages');
+  }
+
 }
 
-class FictionBook {
-  // your code here
+class FictionBook extends Book{
+  String genre;
+  FictionBook(this.genre) : super('', '', 0);
+  @override
+  void bookDetails(){
+    print('$title, $author, $numberOfPages, $genre');
+  }
 }
 
 mixin JsonSerializable {
-  // your code here
+  String? title;
+  String? author;
+  int? numberOfPages;
+
+  Map <dynamic, dynamic> toJson(){
+    Map map = {'Title' : title, 'Author' : author, 'Number of Pages' : numberOfPages};
+    return map;
+  }
 }
 
-double calculateArea(){
-  // your code here
+double calculateArea(double length, double width, String shape){
+  double area = 0.0;
+  if(shape == 'circle'){
+    area = 3.14 * length * length;
+  }
+  else if (shape == 'rectangle'){
+    area = length * width;
+  }
+  return area;
 }
 
 class User{
-  // your code here
+  String name;
+  int? age;
+  String? email;
+
+  User(this.name, this.age, this.email);
```

Comment:
Positional arguments will be used in constructor anyway, I cannot create your User without mail

Reply:
Name has to me required